src/notifications/use-on-push-notification-opened.ts

```diff
@@ -0,0 +1,55 @@
+import {RootNavigationProps} from '@atb/stacks-hierarchy';
+import {
+  firebase,
+  FirebaseMessagingTypes,
+} from '@react-native-firebase/messaging';
+import {useNavigation} from '@react-navigation/native';
+import {useCallback, useEffect} from 'react';
+import {NotificationPayload, NotificationPayloadType} from './types';
+import Bugsnag from '@bugsnag/react-native';
+
+export function useOnPushNotificationOpened() {
+  const {navigate} = useNavigation<RootNavigationProps>();
+
+  const onMessage = useCallback(
+    (message: FirebaseMessagingTypes.RemoteMessage) => {
+      const isKnownType = Object.values(NotificationPayloadType).includes(
+        message.data?.type as NotificationPayloadType,
+      );
+      if (!isKnownType) return;
+
+      const messageData = message.data as NotificationPayload;
```

Comment:
How about defining `NotificationPayload` as a Zod object and use Zod to safe parse the data, rather than assuming it's as expected?

Reply:
Good point! Updated now.